### PR TITLE
Cleanup Scripts

### PR DESCRIPTION
**Pull request by ric-evans**
closes https://github.com/WIPACrepo/file_catalog/issues/68 and https://github.com/WIPACrepo/file_catalog/issues/73

##### `deduplicate.py`
- script for finding & deleting duplicate File Catalog records (aka evil twins), where one is listed as the absolute path (Ex: rooted at /data/exp/) and the other is the realpath (rooted at /mnt/lfs6/exp/)

##### `cast_season_years.py` 
- script for finding & fixing File Catalog records where the `"offline_processing_metadata.season"` (year) is a string. The "offline_processing_metadata" object is patched with an int-cast season value.

### CODE REVIEW

**Comment by ric-evans** [comment]
relies on https://github.com/WIPACrepo/file_catalog/pull/74

**Comment by ric-evans** [comment]
relies on https://github.com/WIPACrepo/file_catalog/pull/76

**Comment by ric-evans** [comment]
`deduplicate.py` is done

File: resources/cleanup_scripts/deduplicate.py [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+    previous_page: List[Dict[str, Any]] = []
+    page = 0
+    while True:
+        logging.info(
+            f"Looking for more bad-rooted paths (page={page}, limit={PAGE_SIZE})..."
+        )
+
+        # Query
+        body = {"start": page * PAGE_SIZE, "limit": PAGE_SIZE, "all-keys": True}
+        resp = rc.request_seq("GET", "/api/files", body)
```
Comment (dsschult) [inline resources/cleanup_scripts/deduplicate.py:210]: Would it be faster to do the logical name query here, so it happens on the server side?  There should be a way to send a mongo query to the server to ask for only prefixed logical names.

**Review by dsschult — APPROVED** [review]
Generally looks good

File: resources/cleanup_scripts/deduplicate.py (hunk 1 above)
Comment (ric-evans) [reply resources/cleanup_scripts/deduplicate.py:210]: Is that possible?

**Comment by ric-evans** [comment]
`cast_season_years.py` is ready for review

File: resources/cleanup_scripts/deduplicate.py (hunk 1 above)
Comment (dsschult) [reply resources/cleanup_scripts/deduplicate.py:210]:
```python
rc.request_seq('GET', '/api/files', {'limit': 1, 'query':json.dumps({"logical_name":{ '$regex': '^\/mnt\/lfs.*' }})})
```

Comment (ric-evans) [reply resources/cleanup_scripts/deduplicate.py:222]: I didn't know mongo supported regex--that cleans things up nicely. 

Now I'm thinking that adding a "directory" base argument (which would essentially make a `$regex` query on the path name) would be useful.

Comment (dsschult) [reply resources/cleanup_scripts/deduplicate.py:222]: Sounds like another PR :smile:

Comment (ric-evans) [reply resources/cleanup_scripts/deduplicate.py:222]: https://github.com/WIPACrepo/file_catalog/issues/77

**Pull request merged**